packages/coachmark/src/Coachmark.ts

```diff
@@ -13,32 +13,412 @@ governing permissions and limitations under the License.
 import {
     CSSResultArray,
     html,
-    SpectrumElement,
+    nothing,
+    PropertyValues,
     TemplateResult,
 } from '@spectrum-web-components/base';
-import { property } from '@spectrum-web-components/base/src/decorators.js';
-
-import styles from './coachmark.css.js';
+import {
+    property,
+    query,
+} from '@spectrum-web-components/base/src/decorators.js';
+import { when } from '@spectrum-web-components/base/src/directives.js';
+import { LikeAnchor } from '@spectrum-web-components/shared/src/like-anchor.js';
+import coachmarkStyles from './coachmark.css.js';
+import chevronStyles from '@spectrum-web-components/icon/src/spectrum-icon-chevron.css.js';
+import '@spectrum-web-components/icons-ui/icons/sp-icon-chevron200.js';
+import { Popover } from '@spectrum-web-components/popover';
+import { join } from 'lit/directives/join.js';
+import { ifDefined } from '@spectrum-web-components/base/src/directives.js';
+import { unsafeHTML } from 'lit/directives/unsafe-html.js';
+import { Placement } from '@spectrum-web-components/overlay';
+import { MediaType, VideoType } from './CoachmarkItem.js';
+import type { CoachmarkItem } from './CoachmarkItem.js';
+import '@spectrum-web-components/asset/sp-asset.js';
+import '@spectrum-web-components/button/sp-button.js';
+import '@spectrum-web-components/quick-actions/sp-quick-actions.js';
 
 /**
  * @element sp-coachmark
+ * @slot cover-photo - This is the cover photo for Default and Quiet Cards
+ * @slot heading - HTML content to be listed as the heading
+ * @slot description - A description of the card
+ * @slot actions - an `sp-action-menu` element outlining actions to take on the represened object
+ * @slot footer - Footer text
  */
-export class Coachmark extends SpectrumElement {
+export class Coachmark extends LikeAnchor(Popover) {
     public static override get styles(): CSSResultArray {
-        return [styles];
+        return [...super.styles, coachmarkStyles, chevronStyles];
+    }
+    @property({ type: Object })
+    public item?: CoachmarkItem;
+
+    @property({ type: String })
+    public override placement: Placement = 'right';
+
+    @property({ type: Object, attribute: false })
+    private content?: {
+        title?: string;
+        description?: string;
+        imageAlt?: string;
+    };
+
+    @property({ type: Number })
+    public offset = 6;
+
+    @property({ attribute: 'shortcut-key' })
+    private shortcutKey?: string;
+
+    @property({ type: Array })
+    public modifierKeys?: string[] = [];
+
+    @property({ attribute: 'src' })
+    private source?: string;
+
+    @property({ attribute: 'media-type' })
+    private mediaType?: MediaType;
+
+    @property({ attribute: 'video-type' })
+    private videoType?: VideoType;
```

Comment:
I'm not sure we want to do any of the video work in these components since it isn't part of the spec for coachmark. We probably want to use ue-video in the future for this functionality, and since that's a private project, I think it's better we don't do it in Coachmark. 

Do you know who designed [this spec](https://spectrum-contributions.corp.adobe.com/page/coach-mark-beta/) and if there are any product teams looking for this implementation? Maybe that can tell us what we should do.

Reply:
Yea make sense. I assumed we are gonna support every media. Yes I am in talks with the design team. I would prefer to create the placeholder for images and gifs for now. I can add video as an extension later on too.